workspace/src/localization/ground_truth/ground_truth/ground_truth.py

```diff
@@ -0,0 +1,170 @@
+import csv
+import rclpy
+from rclpy.node import Node
+from art_msgs.msg import VehicleState
+from chrono_ros_interfaces.msg import DriverInputs as VehicleInput
+from sensor_msgs.msg import Imu, NavSatFix, MagneticField
+from chrono_ros_interfaces.msg import Body as ChVehicle
+import matplotlib.pyplot as plt
+import matplotlib
+import math
+import numpy as np
+import sys
+import os
+from enum import Enum
+from localization_shared_utils import get_dynamics, get_coordinate_transfer
+
+
+class GroundTruthNode(Node):
+    def __init__(self):
+        super().__init__("state_estimation_node")
+
+        # ROS PARAMETERS
+        self.use_sim_msg = (
+            self.get_parameter("use_sim_time").get_parameter_value().bool_value
+        )
+
+        # update frequency of this node
+        self.freq = 10.0
+
+        self.gps = ""
+        self.ground_truth = ""
+        self.mag = ""
+
+        # x, y, from measurements
+        self.x = 0
+        self.y = 0
+
+        # what we will be using for our state vector. (x, y, theta yaw, v vel)
+        self.state = np.zeros((4, 1))
+        np.vstack(self.state)
```

Comment:
how is this file different from particle filter?

Reply:
publisher